pkg/crc/machine/vsock.go

```diff
@@ -43,6 +41,29 @@ func isOpened(exposed []types.ExposeRequest, port types.ExposeRequest) bool {
 	return false
 }
 
+func unExposePorts() error {
+	daemonClient, alreadyOpenedPorts, err := getClientAndOpenPorts()
+	if err != nil {
+		return err
+	}
+	for _, port := range alreadyOpenedPorts {
+		if err := daemonClient.NetworkClient.Unexpose(&types.UnexposeRequest{Local: port.Local}); err != nil {
+			return errors.Wrapf(err, "failed to unexpose port %s ", port.Local)
```

Comment:
Might be a good opportunity for using `Multierror`, we want to unexpose as many ports as possible rather than stop at the first failure.

Reply:
@cfergeau I think we should return as soon as any port which is exposed and api not able to unexpose it. Even if we use Multierror, at the end we are failing the delete operation because of an error occurred.

Reply:
After a `crc delete`, we want to be able to run `crc start`. Even there are some failures, we should try to delete as much as possible. If a port unexpose fails, then we should still try to unexpose as many ports as possible. If there was a failure during port unexpose, imo we should still do the rest of the delete process and remove the VM/...
Especially as there seems to be a small race we cannot do much about (`crc delete` starts unexposing ports, but podman-machine unexposed one of the ports while `crc delete` was running).